src/main/java/server/HTTPServer.java

```diff
@@ -1,37 +1,36 @@
 package server;
 
 import java.io.IOException;
-import java.util.ArrayList;
 
 public class HTTPServer implements Application {
 
     private final String protocol;
+    private final RequestParsable requestParser;
 
-    public HTTPServer() {
+    public HTTPServer(RequestParsable requestParser) {
         this.protocol = "HTTP/1.1";
+        this.requestParser = requestParser;
     }
 
     @Override
     public void call(ReadableWriteable readerWriter) throws IOException {
-        ArrayList<String> request = readRequest(readerWriter);
-        String path = getPath(request);
-        Response response = route(path);
+        Request request = requestParser.call(readRequest(readerWriter));
+        Response response = route(getPath(request));
         response.write(readerWriter);
     }
 
-    private ArrayList<String> readRequest(ReadableWriteable readerWriter) throws IOException {
-        String line = readerWriter.readLine();
-        ArrayList<String> request = new ArrayList<>();
+    private String getPath(Request request) {
+        return request.requestLine().split("\\s")[1];
+    }
 
-        request.add(line);
+    private String readRequest(ReadableWriteable readerWriter) throws IOException {
```

Comment:
This method currently cannot handle requests that have multiple headers. The RequestParser uses the CRLFs to separate the request line from the headers and the body, but BufferedReader.readline() removes the whitespace characters. I'm looking into ways to avoid removing whitespace, but any suggestions are welcome!